Formula/acton.rb

```diff
@@ -0,0 +1,43 @@
+class Acton < Formula
+  desc "Safe actor-based programming language"
+  homepage "https://www.acton-lang.org"
+  url "https://github.com/actonlang/acton/archive/refs/tags/v0.7.1.tar.gz"
+  sha256 "e58487e8c5b17a669080714a2b9940e02f2e26c86e7a549b0fdcbde92ea866ae"
+  license "BSD-3-Clause"
+  head "https://github.com/actonlang/acton.git", branch: "main"
+
+  depends_on "haskell-stack" => :build
+
+  depends_on "protobuf-c"
+  depends_on "utf8proc"
+  depends_on "util-linux"
+
+  on_macos do
+    depends_on "argp-standalone" => :build
+  end
+
+  on_linux do
+    depends_on "gcc"
```

Comment:
Why do we need this? Note that adding `depend_on "gcc"` is not enough to make `brew` use Homebrew GCC as the C/C++ compiler. (It'll be used for `gfortran`, though.)

Reply:
Acton is a programming language. It uses C as an intermediate language and to compile the final binary executable we need a C compiler. Acton uses `cc`, so on Mac it is clang while on the typical Linux machine, it is gcc. On Mac we get clang by default so no need to declare a dependency while on Linux I think we must declare the dependency, no?

I've seen there is some statement for when using parts of the base system too, maybe that is appropriate to use here as well?

Reply:
The situation is the same Linux: `brew` uses the host compiler (i.e. GCC) which is a requirement for using `brew` on Linux. See: https://docs.brew.sh/Homebrew-on-Linux#requirements